src/core/Env.js

```diff
@@ -1090,6 +1090,15 @@ getJasmineRequireObj().Env = function(j$) {
       return spec;
     };
 
+    this.setProperty = function(key, value) {
+      if (!currentRunnable()) {
+        throw new Error(
+          "'setProperty' was used when there was no current spec"
+        );
+      }
+      currentRunnable().setProperty(key, value);
```

Comment:
I haven't tried it out, but I think this will throw if the current runnable is a suite, such as if `setProperty` is called from a `beforeEach`. Line 1094 should probably check that the current runnable is a spec and not just anything truthy, but I could also see a case for allowing suites to have properties.

Reply:
I implemented setSuiteProperty().

I did not find a way to prevent a call to setSuiteProperty() from within a spec, but I think this is minor.

Reply:
It could be done by checking that `currentSpec` is null. But I'm not convinced that it's necessary.